sections/Chapter 3 Sprint 1/postgreSQL.tex

```diff
@@ -2,33 +2,92 @@ \subsection{Relational Databases}
     \item Isolation: Transactions are unaware of other transactions being executed concurrently to avoid confusion.
     \item Durability: Changes caused by a committed transaction persist even in the event of system failures.
 \end{itemize}
 
-A relational database consists of several layers.
-The lowest level is the physical layer which describes how the data are stored physically.
-The purpose of the relational model is abstract over the physical layer of the database.
+In the coming sections, we will describe how one can model and describe a relational database design, both mathematically and with a more graphic design.
+After doing this, we will examine how operations on these models can be described using SQL (Section~\ref{sec:SQL}) and relational algebra (Section~\ref{sec:relationalAlgebra}).
 
-This abstraction is known as the logical layer and allows database administrators to manage the physical storage without directly manipulating the physical data representation.
-The logical layer describes what data are stored and the relationships between the data.
+\subsubsection*{Mathematical description}
+Relational database systems can be mathematically described using relations and sets, mapping a unique key to a tuple of information \cite[Chapter~2.3]{DBSBook}.
+The values of the tuples contained within the relation can be described by the attributes of the relation and their corresponding domains \cite{KatjaFirstPP}. 
+The relations are often described using a \textit{relational schema}, denoting the name and domains of the attributes.
 
-The highest of abstraction is the view layer which describes only part of the database. It exits to simplify the interaction with the system. Many views may exist for the same database.
+Equation \ref{eq:relational_schema} shows an example of a relation describing books as tuples containing three text fields and a positive integer.
+The relation also denotes a super key for the relation. A super key is one or more attributes that can uniquely identify a tuple in a given relation. Attributes describing the superkey are underlined.
 
-Entity-relation diagrams (ER diagrams) can be used to express the relationships in a relational database in a concise and graphical manner \cite{DBSBook} \cite{OracleRDBMS}.
\ No newline at end of file
+\begin{equation} \label{eq:relational_schema}
+    book(author\_name:text, title: text, number\_of\_pages:\mathbb{Z}^+, \underline{ISBN: text})
+\end{equation}
+Superkeys can be defined as $t_1 \in r,\neq t_2 \in r \implies t_1.K \neq t_2.K$. 
+
+That is, no two tuples $t_1, t_2$ from relation $r$ have same values for all superkey attributes $K$. 
+If the superkey does not contain extraneous attributes it is said to be \textit{minimal}. \cite[Chapter 2.3]{DBSBook}
+We will use the term \textit{primary key} to denote a chosen superkey of a relation. 
```

Comment:
```suggestion
We will use the term \textit{primary key} to denote a chosen super key of a relation. 
```

Reply:
Should probably elaborate here what you mean by "chosen", i.e. that it is chosen by the database designers themselves.